pkg/prometheus/promcfg.go

```diff
@@ -1270,17 +1286,22 @@ func (cg *ConfigGenerator) generateServiceMonitorConfig(
 		}
 	}
 
+	sourceLabels := []string{"__meta_kubernetes_endpoint_address_target_kind", "__meta_kubernetes_endpoint_address_target_name"}
+	if cg.EndpointSliceSupported() {
+		sourceLabels = []string{"__meta_kubernetes_endpointslice_address_target_kind", "__meta_kubernetes_endpointslice_address_target_name"}
+	}
+
 	// Relabel namespace and pod and service labels into proper labels.
 	relabelings = append(relabelings, []yaml.MapSlice{
 		{ // Relabel node labels for pre v2.3 meta labels
```

Comment:
This isn't your fault but the comment is misleading :)
could you fix it?

```suggestion
		{ // Relabel node labels with meta labels available with Prometheus >= v2.3.
```